game-core/src/test/java/games/strategy/debug/error/reporting/ErrorReportUploadActionTest.java

```diff
@@ -0,0 +1,135 @@
+package games.strategy.debug.error.reporting;
+
+import static java.util.Arrays.asList;
+import static java.util.Arrays.stream;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+import javax.swing.JFrame;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.triplea.http.client.SendResult;
+import org.triplea.http.client.ServiceClient;
+import org.triplea.http.client.ServiceResponse;
+import org.triplea.http.client.error.report.create.ErrorReport;
+import org.triplea.http.client.error.report.create.ErrorReportResponse;
+
+@ExtendWith(MockitoExtension.class)
+class ErrorReportUploadActionTest {
+
+
+  @Mock
+  private ServiceClient<ErrorReport, ErrorReportResponse> serviceClient;
+
+  @Mock
+  private ConfirmationDialogController dialogController;
+
+  @InjectMocks
+  private ErrorReportUploadAction errorReportUploadAction;
+
+
+  @Mock
+  private JFrame frame;
+
+  private static final UserErrorReport USER_ERROR_REPORT = UserErrorReport.builder()
+      .build();
+
+
+  @Test
+  void nonSuccessSends() {
+    withNonSuccessSendResults()
+        .forEach(notSuccessfulSend -> {
+          when(serviceClient.apply(USER_ERROR_REPORT.toErrorReport()))
+              .thenReturn(notSuccessfulSend);
+
+          errorReportUploadAction.accept(null, USER_ERROR_REPORT);
+
+          verify(dialogController).showFailureConfirmation(notSuccessfulSend);
+
+          reset(serviceClient, dialogController);
```

Comment:
You might want to consider using JUnit's `@ParameterizedTest` in order to remove the `reset()` smells:

```diff
diff --git a/build.gradle b/build.gradle
index fb23dd8f7..2203db01a 100644
--- a/build.gradle
+++ b/build.gradle
@@ -76,6 +76,7 @@ subprojects {
         testImplementation 'nl.jqno.equalsverifier:equalsverifier:3.1.4'
         testImplementation "org.hamcrest:java-hamcrest:$hamcrestVersion"
         testImplementation "org.junit.jupiter:junit-jupiter-api:$junitJupiterVersion"
+        testImplementation "org.junit.jupiter:junit-jupiter-params:$junitJupiterVersion"
         testImplementation 'org.junit-pioneer:junit-pioneer:0.3.0'
         testImplementation "org.mockito:mockito-core:$mockitoVersion"
         testImplementation "org.mockito:mockito-junit-jupiter:$mockitoVersion"
diff --git a/game-core/src/test/java/games/strategy/debug/error/reporting/ErrorReportUploadActionTest.java b/game-core/src/test/java/games/strategy/debug/error/reporting/ErrorReportUploadActionTest.java
index 8efca31c8..786c093d3 100644
--- a/game-core/src/test/java/games/strategy/debug/error/reporting/ErrorReportUploadActionTest.java
+++ b/game-core/src/test/java/games/strategy/debug/error/reporting/ErrorReportUploadActionTest.java
@@ -2,7 +2,6 @@ package games.strategy.debug.error.reporting;
 
 import static java.util.Arrays.asList;
 import static java.util.Arrays.stream;
-import static org.mockito.Mockito.reset;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
@@ -13,6 +12,8 @@ import javax.swing.JFrame;
 
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
@@ -43,22 +44,19 @@ class ErrorReportUploadActionTest {
       .build();
 
 
-  @Test
-  void nonSuccessSends() {
-    withNonSuccessSendResults()
-        .forEach(notSuccessfulSend -> {
-          when(serviceClient.apply(USER_ERROR_REPORT.toErrorReport()))
-              .thenReturn(notSuccessfulSend);
-
-          errorReportUploadAction.accept(null, USER_ERROR_REPORT);
+  @ParameterizedTest
+  @MethodSource("withNonSuccessSendResults")
+  void nonSuccessSends(final ServiceResponse<ErrorReportResponse> notSuccessfulSend) {
+    when(serviceClient.apply(USER_ERROR_REPORT.toErrorReport()))
+        .thenReturn(notSuccessfulSend);
 
-          verify(dialogController).showFailureConfirmation(notSuccessfulSend);
+    errorReportUploadAction.accept(null, USER_ERROR_REPORT);
 
-          reset(serviceClient, dialogController);
-        });
+    verify(dialogController).showFailureConfirmation(notSuccessfulSend);
   }
 
-  private Collection<ServiceResponse<ErrorReportResponse>> withNonSuccessSendResults() {
+  @SuppressWarnings("unused") // called via reflection
+  private static Collection<ServiceResponse<ErrorReportResponse>> withNonSuccessSendResults() {
     // Create ServiceResponses with each of the 'SendResult' values except for the successful SendResult value: 'SENT'
     return stream(SendResult.values())
         .filter(sendResult -> sendResult != SendResult.SENT)
@@ -72,22 +70,18 @@ class ErrorReportUploadActionTest {
         .build();
   }
 
-  @Test
-  void missingIssueLinkInPaylaod() {
-    withMissingIssueLinkResult()
-        .forEach(missingLinkResult -> {
-
-          when(serviceClient.apply(USER_ERROR_REPORT.toErrorReport()))
-              .thenReturn(missingLinkResult);
-
-          errorReportUploadAction.accept(null, USER_ERROR_REPORT);
+  @ParameterizedTest
+  @MethodSource("withMissingIssueLinkResult")
+  void missingIssueLinkInPaylaod(final ServiceResponse<ErrorReportResponse> missingLinkResult) {
+    when(serviceClient.apply(USER_ERROR_REPORT.toErrorReport()))
+        .thenReturn(missingLinkResult);
 
-          verify(dialogController).showFailureConfirmation(missingLinkResult);
+    errorReportUploadAction.accept(null, USER_ERROR_REPORT);
 
-          reset(serviceClient, dialogController);
-        });
+    verify(dialogController).showFailureConfirmation(missingLinkResult);
   }
 
+  @SuppressWarnings("unused") // called via reflection
   private static Collection<ServiceResponse<ErrorReportResponse>> withMissingIssueLinkResult() {
     return asList(
         ServiceResponse.<ErrorReportResponse>builder()
```

Reply:
27c6fccfc

Of note, the `//suppressWarnings` was not needed. There was even a warning when when the method name was incorrect in the annotation (notice how a bunch of 'L' characters were appended to the method name in annotation, mangling it) :
![screenshot from 2019-01-09 10-45-48](https://user-images.githubusercontent.com/12397753/50921140-24b46800-13fc-11e9-9640-23f0f1ffe366.png)

@ssoloff is the suppress warnings needed from what you can see?

Reply:
The warning suppression is required in Eclipse.  However, I haven't tested the latest 2018-12 version yet, but I suspect there is no deeper JUnit integration with the static analyzer.

Reply:
No objection here to add the suppression in the meantime, done:  83527f827